sdks/python/apache_beam/io/gcp/bigtableio_it_test.py

```diff
@@ -0,0 +1,189 @@
+      for j in range(COLUMN_COUNT):
+        test_row.set_cell(column_family_id=COLUMN_FAMILY_ID,
+                          column=('field%s' % j).encode('utf-8'),
+                          value=value,
+                          timestamp=datetime.datetime.now())
+      yield test_row
+
+  def expand(self, pvalue):
+    return (pvalue
+            | beam.Create(self._generate())
+            | bigtableio.WriteToBigTable(project_id=self.beam_options['project_id'],
+                                         instance_id=self.beam_options['instance_id'],
+                                         table_id=self.beam_options['table_id']))
+
+@unittest.skipIf(Client is None, 'GCP Bigtable dependencies are not installed')
+class BigtableIOTest(unittest.TestCase):
+  """ Bigtable IO Connector Test
+
+  This tests the connector both ways, first writing rows to a new table, then reading them and comparing the counters
+  """
+  def setUp(self):
+    self.result = None
+    self.table = Client(project=PROJECT_ID, admin=True)\
+                    .instance(instance_id=INSTANCE_ID)\
+                    .table(TABLE_ID)
+
+    if not self.table.exists():
+      column_families = {COLUMN_FAMILY_ID: column_family.MaxVersionsGCRule(2)}
+      self.table.create(column_families=column_families)
+      logging.info('Table {} has been created!'.format(TABLE_ID))
+
+  @attr('IT')
+  def test_bigtable_io(self):
+    print 'Project ID: ', PROJECT_ID
+    print 'Instance ID:', INSTANCE_ID
+    print 'Table ID:   ', TABLE_ID
+
+    pipeline_options = PipelineOptions(pipeline_parameters(job_name=make_job_name()))
+    p = beam.Pipeline(options=pipeline_options)
+    _ = (p | 'Write Test Rows' >> GenerateTestRows())
+
+    self.result = p.run()
+    self.result.wait_until_finish()
+
+    assert self.result.state == PipelineState.DONE
+
+    if not hasattr(self.result, 'has_job') or self.result.has_job:
+      query_result = self.result.metrics().query(MetricsFilter().with_name('Written Row'))
+      if query_result['counters']:
+        read_counter = query_result['counters'][0]
+        logging.info('Number of Rows written: %d', read_counter.committed)
+        assert read_counter.committed == ROW_COUNT
+
+    pipeline_options = PipelineOptions(pipeline_parameters(job_name=make_job_name('read')))
+    p = beam.Pipeline(options=pipeline_options)
+    count = (p
+             | 'Read from Bigtable' >> bigtableio.ReadFromBigTable(project_id=PROJECT_ID,
+                                                                   instance_id=INSTANCE_ID,
+                                                                   table_id=TABLE_ID,
+                                                                   filter_=b'')
+             | 'Count Rows' >> Count.Globally())
+    self.result = p.run()
+    self.result.wait_until_finish()
+    assert_that(count, equal_to([ROW_COUNT]))
+
+
+if __name__ == '__main__':
```

Comment:
This whole setup is not necessary. See how a class for an integration test defines a test_pipeline in the setup method:

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery_test.py#L485-L507

And then, see how that test_pipeline is used in the test case:

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery_test.py#L548-L568

No more special setup is needed in `__main__`, as it's all done by the testing framework:

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery_test.py#L676-L678

The way you use `assert_that` is correct, so that should work here.